pkg/server/plugin/datastore/datastore.go

```diff
@@ -14,9 +14,9 @@ type DataStore interface {
 	// Bundles
 	AppendBundle(context.Context, *AppendBundleRequest) (*AppendBundleResponse, error)
 	CountBundles(context.Context) (int32, error)
-	CreateBundle(context.Context, *CreateBundleRequest) (*CreateBundleResponse, error)
-	DeleteBundle(context.Context, *DeleteBundleRequest) (*DeleteBundleResponse, error)
-	FetchBundle(context.Context, *FetchBundleRequest) (*FetchBundleResponse, error)
+	CreateBundle(context.Context, *common.Bundle) (*common.Bundle, error)
+	DeleteBundle(ctx context.Context, trustDomain string, mode DeleteMode) error
+	FetchBundle(ctx context.Context, trustDomain string) (*common.Bundle, error)
```

Comment:
When not using the `spiffeid.TrustDomain` type (which we'll eventually want to move this code to use), we need to be specific about what is in that string. `trustDomain` is ambiguous. Is it the trust domain name, e.g. `example.org`, or is it the trust domain ID, e.g. `spiffe://example.org`. The datastore layer operates on trust domain ID. So if we keep this as a string, and not the `spiffeid.TrustDomain` type, then I think we should name thes variables `trustDomainID`.

Reply:
I'm not necessarily advocating we move to `spiffeid.TrustDomain` right now. There are other datastore APIs that we'll also want to change over to `spiffeid.ID`, but not yet, since we're having discussions in SIG-SPEC that impacts the implementation of the go-spiffe v2 ID and trust domain types and we'll need to carefully audit their usage after making changes to ensure behavior does not change.

Reply:
Great point, I did do some looking around during the refactor to see the usage of trustDomain, trustDomainID, and spiffeid.TrustDomain. I'll open an issue to standardize spiffeid datatypes in the datastore.